saleor/graphql/product/types/products.py

```diff
@@ -1428,6 +1443,45 @@ def map_channel_context(variants):
 
         return variants.then(map_channel_context)
 
+    @staticmethod
+    def resolve_product_variants(root: ChannelContext[models.Product], info, **kwargs):
+        requestor = get_user_or_app_from_context(info.context)
+        has_required_permissions = has_one_of_permissions(
+            requestor, ALL_PRODUCTS_PERMISSIONS
+        )
+        channel = root.channel_slug
+        limited_channel_access = False if root.channel_slug is None else True
+        if channel is None and not has_required_permissions:
+            channel = str(
+                get_default_channel_slug_or_graphql_error(
+                    allow_replica=info.context.allow_replica
+                )
+            )
+
+        def _resolve_product_variants(channel_obj):
+            qs = resolve_product_variants(
+                info,
+                channel=channel_obj,
+                product_id=root.node.pk,
+                limited_channel_access=limited_channel_access,
+                requestor=requestor,
+            )
+            kwargs["channel"] = qs.channel_slug
+            qs = filter_connection_queryset(
+                qs, kwargs, allow_replica=info.context.allow_replica
+            )
+            return create_connection_slice(
+                qs, info, kwargs, ProductVariantCountableConnection
+            )
+
+        if channel:
+            return (
+                ChannelBySlugLoader(info.context)
+                .load(channel)
+                .then(_resolve_product_variants)
+            )
+        return _resolve_product_variants(None)
```

Comment:
Looks like the test is missing for calling `_resolve_product_variants` with `None`